tests/asyncio_test_helpers.py

```diff
@@ -0,0 +1,176 @@
+import logging
+import types
+from asyncio import AbstractEventLoop, transports
+from asyncio.protocols import BaseProtocol, Protocol
+from asyncio.transports import Transport
+from typing import Any, Callable, List, Tuple
+
+logger = logging.getLogger(__name__)
+
+
+class TimelessEventLoopWrapper:
+    @property  # type: ignore
+    def __class__(self):
+        """
+        Fakes isinstance(this, AbstractEventLoop) so we can set_event_loop
+        without fail.
+        """
+        return self._wrapped_loop.__class__
```

Comment:
Is this here to make our own assertion pass or will something deep in asyncio complain without this?

Reply:
asyncio's `set_event_loop` will complain without this.